packages/jac/src/json-api.test.ts

```diff
@@ -127,6 +137,70 @@ describe('performs JsonApiClient request unit test', () => {
     })
   })
 
+  test('should throw network error', () => {
+    const rawErrorResponse = MockWrapper.makeFetcherResponse(
+      {
+        errors: [
+          {
+            id: '1',
+            code: 'err_some_code',
+            status: '405',
+            title: 'Unhandled error',
+          },
+        ],
+      },
+      405,
+    )
+
+    const api = new JsonApiClient(VALID_CFG)
+
+    jest
+      .spyOn(Fetcher.prototype, 'request')
+      .mockRejectedValue(new FetcherError(rawErrorResponse))
+
+    expect(api.get('')).rejects.toThrowError(errors.NetworkError)
+  })
+
+  test('should throw unauthorized error', () => {
+    const rawErrorResponse = MockWrapper.makeFetcherResponse(
+      {
+        errors: [
+          {
+            id: '1',
+            code: 'err_some_code',
+            status: '401',
+            title: 'Unauthorized',
+          },
+        ],
+      },
+      401,
+    )
+
+    const api = new JsonApiClient(VALID_CFG)
+
+    jest
+      .spyOn(Fetcher.prototype, 'request')
+      .mockRejectedValue(new FetcherError(rawErrorResponse))
+
+    expect(api.get('')).rejects.toThrowError(errors.UnauthorizedError)
+  })
```

Comment:
this cases have already covered there https://github.com/distributed-lab/web-kit/blob/main/packages/jac/src/middlewares/parse-json-api-error.test.ts